src/utils/navigatorData.js

```diff
@@ -179,16 +179,21 @@ export function getSiblings(uid, childrenMap, children) {
 }
 
 function extractRootNode(data) {
+  // note: this "root" path won't always necessarily come at the beginning of
+  // the URL in situations where the renderer is being hosted at some path
+  // prefix
+  const rootPathPattern = /^.*(\/documentation\/[^/]+).*$/;
+  const rootPath = window.location.href.match(rootPathPattern)?.[1] ?? '';
   // most of the time, it is expected that `data` always has a single item
   // that represents the top-level root node of the navigation tree
   //
   // there may be rare, unexpected scenarios where multiple top-level root
-  // nodes are provide for some reason—if that happens, we would prefer the
-  // first one categorized as a "module"
+  // nodes are provide for some reason—if that happens, we would prefer the one
+  // with a path that most closely resembles the current URL path
   //
   // otherwise, the first provided node will be used
   return data.length === 1 ? data[0] : (data.find(node => (
-    node.type === TopicTypes.module
+    node.path.toLowerCase() === rootPath.toLowerCase()
```

Comment:
Are we confident we won’t encounter a case where one node’s `path` is `/documentation/foo` and another’s is `/documentation/foo/bar`, and we have to pick the most specific (i.e. longest) one?

Reply:
I might be misunderstanding your concern here, but I can't think of a scenario where a nested item like `/documentation/foo/bar` would be provided as a root level node since the nesting is already visible in the path itself, so that particular node should never be a part of this logic which only runs for the top-level items (and only when there are multiple).